sdks/python/apache_beam/runners/dataflow/internal/dependency.py

```diff
@@ -294,249 +160,26 @@ def stage_job_resources(
-    staging_location: A GCS bucket where the SDK file should be copied.
-    temp_dir: path to temporary location where the file should be downloaded.
-
-  Returns:
-    A list of SDK files that were staged to the staging location.
-
-  Raises:
-    RuntimeError: if staging was not successful.
-  """
-  if (sdk_remote_location.startswith('http://') or
-      sdk_remote_location.startswith('https://')):
-    local_download_file = _dependency_file_download(
-        sdk_remote_location, temp_dir)
-    staged_name = _desired_sdk_filename_in_staging_location(local_download_file)
-    staged_path = FileSystems.join(staging_location, staged_name)
-    logging.info(
-        'Staging Beam SDK from %s to %s',
-        sdk_remote_location, staged_path)
-    _dependency_file_copy(local_download_file, staged_path)
-    return [staged_name]
-  elif sdk_remote_location.startswith('gs://'):
-    # Stage the file to the GCS staging area.
-    staged_name = _desired_sdk_filename_in_staging_location(sdk_remote_location)
-    staged_path = FileSystems.join(staging_location, staged_name)
-    logging.info(
-        'Staging Beam SDK from %s to %s',
-        sdk_remote_location, staged_path)
-    _dependency_file_copy(sdk_remote_location, staged_path)
-    return [staged_name]
-  elif sdk_remote_location == 'pypi':
-    sdk_local_file = _download_pypi_sdk_package(temp_dir)
-    sdk_sources_staged_name = _desired_sdk_filename_in_staging_location(
-        sdk_local_file)
-    staged_path = FileSystems.join(staging_location, sdk_sources_staged_name)
-    logging.info('Staging SDK sources from PyPI to %s', staged_path)
-    _dependency_file_copy(sdk_local_file, staged_path)
-    staged_sdk_files = [sdk_sources_staged_name]
-    try:
-      # Stage binary distribution of the SDK, for now on a best-effort basis.
-      sdk_local_file = _download_pypi_sdk_package(temp_dir, fetch_binary=True)
-      sdk_binary_staged_name = _desired_sdk_filename_in_staging_location(
-          sdk_local_file)
-      staged_path = FileSystems.join(staging_location, sdk_binary_staged_name)
-      logging.info('Staging binary distribution of the SDK from PyPI to %s',
-                   staged_path)
-      _dependency_file_copy(sdk_local_file, staged_path)
-      staged_sdk_files.append(sdk_binary_staged_name)
-    except RuntimeError as e:
-      logging.warn('Failed to download requested binary distribution '
-                   'of the SDK: %s', repr(e))
-
-    return staged_sdk_files
-  else:
-    raise RuntimeError(
-        'The --sdk_location option was used with an unsupported '
-        'type of location: %s' % sdk_remote_location)
+  file_handler = DataflowFileHandle()
+  file_handler.file_copy = file_copy if file_copy else file_handler.file_copy
+  file_handler.file_download = (
+      file_download if file_download else file_handler.file_download)
+  resource_stager = DataFlowStager(file_handler=file_handler)
```

Comment:
<!--new_thread; commit:d2647e7c0c6001a89be4dc0a673e17d5ce0d951a; resolved:0-->
I don't think we need DataflowFileStager class. 
We can instantiate an object: dataflow_file_stager = stager.Stager(file_handler=DataflowFileHandler())